### PR TITLE
Ignore collectionFormat when parsing body. Fixes #61.

### LINKED ISSUE #61: collectionFormat shouldn't be applied when parsing body
I have a minimal example:

* Swagger API:

  ```yaml
  swagger: '2.0'

  info:
   title: Dummy example
   version: "1.0.0"

  basePath: /

  paths:
    /foo/:
      post:
        operationId: foo-bar.api/foo
        consumes:
        - application/json
        parameters:
        - name: bar
          in: body
          required: true
          schema:
            type: array
        responses:
          202:
            description: Pet response
  ```

* Handler of `/foo/`:

  ```clojure
  (defn foo [request]
    (-> (r/response {:message (-> request :parameters :body)})
        (r/content-type "application/json")))
  ```

The server crashes when passing a JSON array in the body:

```sh
curl -v -XPOST -H "Content-Type: application/json" --data '{ "bar": [ ] }' 127.0.0.1:3000/foo
```

Log:
```
Nov 18, 2016 2:10:37 PM io.sarnowski.swagger1st.parser invoke
SEVERE: internal server error java.lang.ClassCastException: clojure.lang.PersistentArrayMap cannot be cast to java.lang.CharSequence
java.lang.ClassCastException: clojure.lang.PersistentArrayMap cannot be cast to java.lang.CharSequence
        at clojure.string$split.invoke(string.clj:217)
        at io.sarnowski.swagger1st.parser$eval6808$fn__6809.invoke(parser.clj:153)
        at clojure.lang.MultiFn.invoke(MultiFn.java:233)
        at io.sarnowski.swagger1st.parser$eval6855$fn__6856$fn__6857.invoke(parser.clj:207)
        at io.sarnowski.swagger1st.parser$create_parser$fn__6914.invoke(parser.clj:317)
        at io.sarnowski.swagger1st.parser$parse$fn__6942.invoke(parser.clj:353)
        at clojure.core$map$fn__4553.invoke(core.clj:2624)
        at clojure.lang.LazySeq.sval(LazySeq.java:40)
        at clojure.lang.LazySeq.seq(LazySeq.java:49)
        at clojure.lang.RT.seq(RT.java:507)
        at clojure.core$seq__4128.invoke(core.clj:137)
        at clojure.core.protocols$seq_reduce.invoke(protocols.clj:30)
        at clojure.core.protocols$fn__6506.invoke(protocols.clj:101)
        at clojure.core.protocols$fn__6452$G__6447__6465.invoke(protocols.clj:13)
        at clojure.core$reduce.invoke(core.clj:6519)
        at clojure.core$group_by.invoke(core.clj:6861)
        at io.sarnowski.swagger1st.parser$parse.invoke(parser.clj:355)
        at io.sarnowski.swagger1st.core$chain_handler$chain_handler__7012$fn__7013.invoke(core.clj:33)
        at io.sarnowski.swagger1st.mapper$correlate.invoke(mapper.clj:240)
        at io.sarnowski.swagger1st.core$chain_handler$chain_handler__7012$fn__7013.invoke(core.clj:33)
        at io.sarnowski.swagger1st.discoverer$discover.invoke(discoverer.clj:45)
        at clojure.core$partial$fn__4527.invoke(core.clj:2495)
        at io.sarnowski.swagger1st.core$chain_handler$chain_handler__7012$fn__7013.invoke(core.clj:33)
        at clojure.lang.Var.invoke(Var.java:379)
        at ring.middleware.reload$wrap_reload$fn__1444.invoke(reload.clj:22)
        at ring.middleware.stacktrace$wrap_stacktrace_log$fn__876.invoke(stacktrace.clj:23)
        at ring.middleware.stacktrace$wrap_stacktrace_web$fn__938.invoke(stacktrace.clj:86)
        at ring.adapter.jetty$proxy_handler$fn__97.invoke(jetty.clj:24)
        at ring.adapter.jetty.proxy$org.eclipse.jetty.server.handler.AbstractHandler$ff19274a.handle(Unknown Source)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
        at org.eclipse.jetty.server.Server.handle(Server.java:497)
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:310)
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:257)
        at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:540)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555)
        at java.lang.Thread.run(Thread.java:745)
```

This happens with Swagger1st `0.22.0+`. It turned out the `collectionFormat` is applied with parameters in body, which it shouldn't according to [Swagger 2.0 spec](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md):

> If in is any value other than "body"

### PR DESCRIPTION
**Pull request by cmpitg**
Fixes #61.

### CODE REVIEW

File: src/io/sarnowski/swagger1st/parser.clj [hunk 1]
```diff
@@ -204,7 +204,9 @@
   (let [items-definition (get definition "items")
         items-parser (create-value-parser items-definition path parser-options)]
     (fn [value]
-      (let [vals (if value (split-array definition value) [])
+      (let [vals (if (and value (not= "body" (first path)))
+                   (split-array definition value)
+                   [])
```
Comment (sebastianpoeplau) [inline src/io/sarnowski/swagger1st/parser.clj]: Wouldn't this set `vals` to the empty vector when parsing the body? I guess we'd rather want something like this, wouldn't we?

``` clojure
(cond
  (= "body" (first path)) value
  (some? value) (split-array definition value)
  :else [])
```

Comment (cmpitg) [reply src/io/sarnowski/swagger1st/parser.clj]: Oops. Yes you're right. Sorry about that. I'm fixing it now.

**Comment by sebastianpoeplau** [comment]
👍

**Comment by dryewo** [comment]
@cmpitg would you mind adding some tests to increase code coverage?

**Comment by cmpitg** [comment]
Yessir. I have updated the code. Please review.

**Comment by dryewo** [comment]
👍

**Comment by sebastianpoeplau** [comment]
👍

**Pull request merged**